File: sp/python/pingpong/system.py
# system.py
import asyncio
from tdpnet import StandardSession

class System(StandardSession):
    def __init__(self, config, connect=True):
        super().__init__()
        self.config = config
        if connect:
            asyncio.create_task(self.connect(self.host, self.config.get('security', {}).get('ca')))

    @property
    def host(self):
        return self.config.get('host')

    @property
    def credential(self):
        return self.config.get('credential')

    @property
    def address(self):
        return self.credential.get('address')

    @property
    def peers(self):
        return self.config.get('peers')

    def on_connected(self, timestamp):
        self.signin(self.credential)

    def on_granted(self, data):
        pass

    def on_denied(self, data):
        pass

File: sp/dart/pingpong/system.dart
// system.dart
import 'core/packet.dart'; // Contains the Entity class
import 'core/tdpnet.dart';

class System extends StandardSession {
  final Entity config;

  System(Map<String, dynamic> configMap, {bool connect = true})
      : config = Entity(data: configMap) {
    if (connect) {
      this.connect(host);
      // Note: In a production Dart/Flutter app, you would pass the CA 
      // string from config['security']['ca'] to a SecurityContext.
    }
  }

  String get host => config['host'];
  
  Map<String, dynamic> get credential => 
      Map<String, dynamic>.from(config['credential']);
      
  String get address => credential['address'];
  
  Map<String, dynamic> get peers => 
      Map<String, dynamic>.from(config['peers']);

  @override
  void onConnected(int timestamp) {
    signin(credential);
  }

  void onGranted(dynamic data) {}
  void onDenied(dynamic data) {}
}